### PR TITLE
ROX-12750: Remove deprecated resources

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The set resources used for access control is being simplified. A number of resources has been deprecated.

The goal here is to get rid of the deprecated resources, both in the database and in the codebase (UI, QA, central, DB).
This PR focuses on the QA and UI parts.

## Checklist
- [ ] Investigated and inspected CI test results
~~- [ ] Unit test and regression tests added~~
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI should be enough

### CODE REVIEW

**Review by janisz — APPROVED** [review]

File: ui/apps/platform/src/types/roleResources.ts [hunk 1]
```diff
@@ -27,26 +27,15 @@ export type ResourceName =
...
-    | 'DebugLogs'
-    | 'NetworkGraphConfig'
-    | 'ProbeUpload'
-    | 'ScannerBundle'
-    | 'ScannerDefinitions'
-    | 'SensorUpgradeConfig'
-    | 'ServiceIdentity'
+    // TODO: ROX-13888 Remove Policy, VulnerabilityReports.
+    | 'Policy'
+    | 'Role'
```
Comment (ivan-degtiarenko) [inline ui/apps/platform/src/types/roleResources.ts:39]: Do we have a ticket for replacing `Role` with `Access`?

File: central/main.go [hunk 2]
```diff
@@ -262,9 +262,11 @@ func main() {
 		"	Integration replaces APIToken, BackupPlugins, ImageIntegration, Notifier, and SignatureIntegration\n" +
 		"	Image now also covers ImageComponent\n" +
 		"The following permission resources will be replaced in the upcoming versions:\n" +
+		"	Access will replace Role\n" +
 		"	Administration will replace AllComments, Config, DebugLogs, NetworkGraphConfig, ProbeUpload, ScannerBundle, ScannerDefinitions, SensorUpgradeConfig, and ServiceIdentity\n" +
```
Comment (ivan-degtiarenko) [inline central/main.go:266]: I think this line should be moved to ` have been replaced` section, right?

File: ui/apps/platform/src/types/roleResources.ts (hunk 1 above)
Comment (dhaus67) [inline ui/apps/platform/src/types/roleResources.ts:39]:
```suggestion
    // TODO: ROX-14398 Replace Role with Access
    | 'Role'
```

**Review by dhaus67 — COMMENTED** [review]
I'll leave approving for UI folks, the code in principle looks good to me.

File: central/main.go [hunk 3]
```diff
@@ -262,9 +262,11 @@ func main() {
...
 		"	Compliance will replace ComplianceRuns\n" +
-		"	Cluster will cover ClusterCVE.")
+		"	Cluster will cover ClusterCVE\n" +
```
[the remark this answers is not in the document]
Comment (rhybrillou) [reply central/main.go]: The replacement was both in the "already replaced" and the "will be replaced" blocks.
I'm pulling the startup log adjustment from https://github.com/stackrox/stackrox/pull/5094 here.

File: migrator/migrations/m_176_to_m_177_group_permissions/migration.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	ScannerBundle       = "ScannerBundle"
+	ScannerDefinitions  = "ScannerDefinitions"
+	SensorUpgradeConfig = "SensorUpgradeConfig"
+	ServiceIdentity     = "ServiceIdentity"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: startSeqNum,
+		VersionAfter:   &storage.Version{SeqNum: int32(startSeqNum + 1)},
```
Comment (dashrews78) [inline migrator/migrations/m_176_to_m_177_group_permissions/migration.go:43]: nit: comment for after num?

**Review by dashrews78 — APPROVED** [review]

**Review by vjwilson — APPROVED** [review]
UI changes OK

**Review by pedrottimark — APPROVED** [review]

**Comment by pedrottimark** [comment]
@rhybrillou for failing ui-e2e-tests you can rebase on #5311

**Pull request merged**